sql/core/src/test/resources/tpcds-plan-stability/approved-plans-modified/q65.sf100/explain.txt

```diff
@@ -151,106 +152,110 @@ Functions [1]: [avg(revenue#21)]
 Aggregate Attributes [1]: [avg(revenue#21)#27]
 Results [2]: [ss_store_sk#13, avg(revenue#21)#27 AS ave#28]
 
-(23) BroadcastExchange
+(23) Filter [codegen id : 6]
```

Comment:
We need to investigate why we have an extra Filter now.

Reply:
The reason is: both `PromotePrecision` and `CheckOverflow` do not inherit `NullIntolerant`, so the optimizer rule `InferFiltersFromConstraints` can not infer is not null from the inside attribute.

It affects all decimal binary arithmetic. This pr removes the`PromotePrecision` and `CheckOverflow` so that we have an extra filter now.

Reply:
I see, and it's actually beneficial.